### ISSUE TITLE
7.1.0 fails to build on windows 7 x86_64

### ISSUE THREAD

**Issue by FROGGS** [parrot/parrot#1203]
```
Configuring Parrot ...
C:\Perl64\bin\perl.exe Configure.pl --icushared="C:\icu\lib64\icudt.lib C:\icu\lib64\icuuc.lib" --icuheaders=C:\icu\include --optimize --prefix=C:/nqp/install
Parrot Version 7.1.0 Configure 2.0
Copyright (C) 2001-2015, Parrot Foundation.

Hello, I'm Configure. My job is to poke and prod your system to figure out
how to build Parrot. The process is completely automated, unless you passed in
the `--ask' flag on the command line, in which case I'll prompt you for a few
pieces of info.

Since you're running this program, you obviously have Perl 5--I'll be pulling
some defaults from its configuration.

init::manifest -      Check MANIFEST.....................................done.
init::defaults -      Set Configure's default values.....................done.
init::install -       Set up installation paths..........................done.
init::hints -         Load platform and local hints files................done.
inter::progs -        Determine what C compiler and linker to use...Der Befehl ".\test_216.exe" ist entweder falsch geschrieben oder konnte nicht gefunden werden.
.....done.
inter::make -         Is make installed...................................yes.
inter::lex -          Is lex installed................................skipped.
inter::yacc -         Is yacc installed...............................skipped.
auto::gcc -           Is your C compiler actually gcc...Der Befehl ".\test_216.exe" ist entweder falsch geschrieben oder konnte nicht gefunden werden.

step auto::gcc died during execution: Can't run the test program:  at config/auto/gcc.pm line 42.

 at Configure.pl line 96.

auto::glibc -         Is GNU libc installed...Der Befehl ".\test_216.exe" ist entweder falsch geschrieben oder konnte nicht gefunden werden.

step auto::glibc died during execution: Can't run the test program:  at config/auto/glibc.pm line 47.

 at Configure.pl line 96.

auto::backtrace -     Does libc have the backtrace* functions..............no.
auto::msvc -          Is your C compiler actually Visual C++...Der Befehl ".\test_216.exe" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.

step auto::msvc died during execution: Can't run the test program:  at config/auto/msvc.pm line 49.

 at Configure.pl line 96.

auto::attributes -    Detect compiler attributes.........................done.
auto::arch -          Determine CPU architecture and type, and OS........done.
auto::cpu -           Generate CPU specific stuff........................done.
init::optimize -      Enable optimization.................................yes.
inter::shlibs -       Determine flags for building shared libraries......done.
inter::libparrot -    Should parrot link against a shared library...
Warning: Building a shared parrot library may conflict with your previously-installed C:\nqp\install\bin\libparrot.dll

Warning: Building a shared parrot library may conflict with your previously-installed C:\nqp\install\lib\libparrot.lib

Warning: Building a shared parrot library may conflict with your previously-installed C:\nqp\install\bin\libparrot.lib
......yes.
inter::types -        What types should Parrot use.......................done.
auto::ops -           Which opcode files should be compiled in...........done.
auto::pmc -           Which pmc files should be compiled in...Terminating on signal SIGINT(2)
Terminating on signal SIGINT(2)
^C
```
The german message `Der Befehl ".\test_216.exe" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.` means `Command ".\test_216.exe" not found` basically.

**Comment by Util** [comment]
Would you please try again, appending these flags to the Configure.pl, and posting the output?
    --verbose=2 --fatal

My first guess is that the libparrot in your installed NQP is causing the problem.

**Comment by FROGGS** [comment]
```
C:\nqp>perl Configure.pl --backends=parrot --gen-parrot --parrot-option="--verbose=2 --fatal --icushared=\"C:\icu\lib64\icudt.lib C:\icu\lib64\icuuc.lib\" --icuheaders=C:
\icu\include" --prefix=C:\nqp\install --make-install
Der Befehl ""C:/nqp/install/bin/parrot.exe"" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
HEAD is now at 2f99a83... Release 7.1.0 - Lilian's Lovebird

Configuring Parrot ...
C:\Perl64\bin\perl.exe Configure.pl --verbose=2 --fatal --icushared="C:\icu\lib64\icudt.lib C:\icu\lib64\icuuc.lib" --icuheaders=C:\icu\include --optimize --prefix=C:/nqp
/install
Parrot Version 7.1.0 Configure 2.0
Copyright (C) 2001-2015, Parrot Foundation.

Hello, I'm Configure. My job is to poke and prod your system to figure out
how to build Parrot. The process is completely automated, unless you passed in
the `--ask' flag on the command line, in which case I'll prompt you for a few
pieces of info.

Since you're running this program, you obviously have Perl 5--I'll be pulling
some defaults from its configuration.

init::manifest -      Check MANIFEST...
Setting Configuration Data:
(
        verbose => '2',
);

                      Check MANIFEST.....................................done.
init::defaults -      Set Configure's default values...
Setting Configuration Data:
(
        clock_best => '',
);

Setting Configuration Data:
(
        archname => 'MSWin32-x64-multi-thread',
);

Setting Configuration Data:
(
        has_core_nci_thunks => '1',
);

Setting Configuration Data:
(
        HAS_CORE_NCI_THUNKS => '1',
);

Setting Configuration Data:
(
        has_extra_nci_thunks => '1',
);

Setting Configuration Data:
(
        HAS_EXTRA_NCI_THUNKS => '1',
);

Setting Configuration Data:
(
        runcore_tests => 't/compilers/imcc/*/*.t t/op/*.t t/pmc/*.t t/oo/*.t t/pir/*.t t/native_pbc/*.t',
);

Setting Configuration Data:
(
        core_tests => 't/src/*.t t/src/embed/*.t t/run/*.t t/perl/*.t',
);

Setting Configuration Data:
(
        library_tests => 't/compilers/pct/*.t t/compilers/pge/*.t t/compilers/pge/p5regex/*.t t/compilers/pge/perl6regex/*.t t/compilers/tge/*.t t/compilers/opsc/*.t t/co
mpilers/data_json/*.t ext/nqp-rx/t/nqp/*.t t/dynoplibs/*.t t/dynpmc/*.t t/library/*.t t/ext/winxed/*.t t/tools/*.t t/profiling/*.t',
);

Setting Configuration Data:
(
        configure_tests => 't/configure/*.t t/steps/*.t t/postconfigure/*.t',
);

Setting Configuration Data:
(
        developing_tests => 't/codingstd/c_arg_assert.t t/codingstd/c_code_coda.t t/codingstd/c_cppcomments.t t/codingstd/c_enum.t t/codingstd/c_function_docs.t t/codings
td/c_header_guards.t t/codingstd/c_indent.t t/codingstd/c_macro_args.t t/codingstd/c_operator.t t/codingstd/c_parens.t t/codingstd/c_returns.t t/codingstd/c_struct.t t/co
dingstd/c_todo.t t/codingstd/case_sensitive_files.t t/codingstd/check_isxxx.t t/codingstd/check_toxxx.t t/codingstd/copyright.t t/codingstd/cuddled_else.t t/codingstd/fil
enames.t t/codingstd/gmt_utc.t t/codingstd/linelength.t t/codingstd/make_code_coda.t t/codingstd/opcode_doc.t t/codingstd/pbc_compat.t t/codingstd/pccmethod_deps.t t/codi
ngstd/pdd_format.t t/codingstd/perlcritic.t t/codingstd/pir_code_coda.t t/codingstd/pmc_docs.t t/codingstd/pod_description.t t/codingstd/pod_syntax.t t/codingstd/pod_todo
.t t/codingstd/tabs.t t/codingstd/trailing_space.t',
);

Setting Configuration Data:
(
        verbose => '2',
);

                      Set Configure's default values.....................done.
init::install -       Set up installation paths...
Setting Configuration Data:
(
        prefix => 'C:/nqp/install',
        exec_prefix => 'C:/nqp/install',
        bindir => 'C:/nqp/install/bin',
        sbindir => 'C:/nqp/install/sbin',
        libexecdir => 'C:/nqp/install/libexec',
        datadir => 'C:/nqp/install/share',
        sysconfdir => 'C:/nqp/install/etc',
        sharedstatedir => 'C:/nqp/install/com',
        localstatedir => 'C:/nqp/install/var',
        libdir => 'C:/nqp/install/lib',
        includedir => 'C:/nqp/install/include',
        oldincludedir => '/usr/include',
        infodir => 'C:/nqp/install/info',
        mandir => 'C:/nqp/install/man',
        srcdir => 'C:/nqp/install/src',
        docdir => 'C:/nqp/install/share/doc',
        versiondir => '/parrot/7.1.0-devel',
);

Setting Configuration Data:
(
        verbose => '2',
);

                      Set up installation paths..........................done.
init::hints -         Load platform and local hints files...
[ init::hints::mswin32
Setting Configuration Data:
(
        win32 => '1',
        PQ => '"',
        make_c => '$(PERL) -e "chdir shift @ARGV; system '$(MAKE)', @ARGV; exit $$? >> 8;"',
        ncilib_link_extra => '-def:src\libnci_test.def',
);

Setting Configuration Data:
(
        clock_best => ' ',
);

Setting Configuration Data:
(
        has_mt => '1',
);

Setting Configuration Data:
(
        cxx => 'cl',
);
 ccflags => '-nologo -GF -W4 -MD -Zi -DNDEBUG -Ox -GL -fp:precise -DWIN32 -D_CONSOLE -DNO_STRICT -DWIN64 -DCONSERVATIVE -DPERL_TEXTMODE_SCRIPTS -DUSE_SITECUSTOMIZE -DPERL
_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO',  libs => 'kernel32.lib ws2_32.lib msvcrt.lib oldnames.lib advapi32.lib ',
Setting Configuration Data:
(
        share_ext => '.dll',
        load_ext => '.dll',
        a => '.lib',
        o => '.obj',
        cc_o_out => '-Fo',
        cc_exe_out => '-out:',
        cc_ldflags => '-link',
        make_c => '$(PERL) -e "chdir shift @ARGV;system '$(MAKE)', '-nologo', @ARGV; exit $$? >> 8;"',
        make => 'nmake',
        cc_debug => '-Zi',
        ld_debug => '-debug',
        ld_share_flags => '-dll',
        ld_load_flags => '-dll',
        ld_out => '-out:',
        ldflags => '-nologo -nodefaultlib -debug -opt:ref,icf -ltcg  -libpath:"C:\Perl64\lib\CORE"  -machine:AMD64',
        linkflags => '-nologo -nodefaultlib -debug -opt:ref,icf -ltcg  -libpath:"C:\Perl64\lib\CORE"  -machine:AMD64',
        libs => 'kernel32.lib ws2_32.lib msvcrt.lib oldnames.lib advapi32.lib ',
        libparrot_static => 'libparrot.lib',
        libparrot_shared => 'libparrot.dll',
        ar => 'lib',
        arflags => '',
        ar_out => '-out:',
        slash => '\',
        ccflags => '-nologo -GF -W4 -MD -Zi -DNDEBUG -Ox -GL -fp:precise -DWIN32 -D_CONSOLE -DNO_STRICT -DWIN64 -DCONSERVATIVE -DPERL_TEXTMODE_SCRIPTS -DUSE_SITECUSTOMIZE
 -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO',
        ccwarn => '-wd4127 -wd4054 -wd4310 ',
        has_dynamic_linking => '1',
        parrot_is_shared => '1',
        sym_export => '__declspec(dllexport)',
        sym_import => '__declspec(dllimport)',
);

Setting Configuration Data:
(
        libparrot_ldflags => '"C:/nqp/parrot\libparrot.lib"',
);

Setting Configuration Data:
(
        libparrot_linkflags => '"C:/nqp/parrot\libparrot.lib"',
);

Setting Configuration Data:
(
        inst_libparrot_ldflags => '"C:/nqp/install/bin\libparrot.lib"',
);

Setting Configuration Data:
(
        inst_libparrot_linkflags => '"C:/nqp/install/bin\libparrot.lib"',
);
init::hints::local ]
Setting Configuration Data:
(
        verbose => '2',
);

                      Load platform and local hints files................done.
inter::progs -        Determine what C compiler and linker to use...
Setting Configuration Data:
(
        cc => 'cl',
);

Setting Configuration Data:
(
        link => 'cl',
);

Setting Configuration Data:
(
        ld => 'link',
);

Setting Configuration Data:
(
        ccflags => '-nologo -GF -W4 -MD -Zi -DNDEBUG -Ox -GL -fp:precise -DWIN32 -D_CONSOLE -DNO_STRICT -DWIN64 -DCONSERVATIVE -DUSE_SITECUSTOMIZE ',
);

ccflags: -nologo -GF -W4 -MD -Zi -DNDEBUG -Ox -GL -fp:precise -DWIN32 -D_CONSOLE -DNO_STRICT -DWIN64 -DCONSERVATIVE -DUSE_SITECUSTOMIZE

Setting Configuration Data:
(
        ar => 'lib',
);

Setting Configuration Data:
(
        arflags => '',
);

Setting Configuration Data:
(
        linkflags => '-nologo -nodefaultlib -debug -opt:ref,icf -ltcg    -machine:AMD64',
);

Setting Configuration Data:
(
        ldflags => '-nologo -nodefaultlib -debug -opt:ref,icf -ltcg    -machine:AMD64',
);

Setting Configuration Data:
(
        libs => 'kernel32.lib ws2_32.lib msvcrt.lib oldnames.lib advapi32.lib',
);

Setting Configuration Data:
(
        cxx => 'cl',
);

Setting Configuration Data:
(
        link_debug => '',
        ld_debug => '',
);

Setting Configuration Data:
(
        cc_debug => '',
);

Setting Configuration Data:
(
        ccwarn => '-wd4127 -wd4054 -wd4310 ',
);

cc=cl, link=cl
cl -nologo -GF -W4 -MD -Zi -DNDEBUG -Ox -GL -fp:precise -DWIN32 -D_CONSOLE -DNO_STRICT -DWIN64 -DCONSERVATIVE -DUSE_SITECUSTOMIZE   -I./include -c test_3108.c
test_3108.c
cl -nologo -nodefaultlib -debug -opt:ref,icf -ltcg    -machine:AMD64 test_3108.obj  -out:test_3108.exe  kernel32.lib ws2_32.lib msvcrt.lib oldnames.lib advapi32.lib
cl : Command line warning D9035 : option 'o' has been deprecated and will be removed in a future release
cl : Command line warning D9035 : option 'o' has been deprecated and will be removed in a future release
cl : Command line warning D9002 : ignoring unknown option '-nodefaultlib'
cl : Command line warning D9002 : ignoring unknown option '-debug'
cl : Command line warning D9002 : ignoring unknown option '-ltcg'
cl : Command line warning D9002 : ignoring unknown option '-machine:AMD64'
test_3108.obj : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
Generating code
Finished generating code
.\test_3108.exe
Der Befehl ".\test_3108.exe" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.

Setting Configuration Data:
(
        verbose => '2',
);

                      Determine what C compiler and linker to use........done.
inter::make -         Is make installed...
Setting Configuration Data:
(
        gmake_version => undef,
);
checking for program: nmake or mingw32-make or gmake or make
C:\Program Files (x86)\Microsoft Visual Studio 11.0\VC\BIN\nmake.EXE is executable

Setting Configuration Data:
(
        make => 'nmake',
);

Setting Configuration Data:
(
        make => 'nmake',
);

Setting Configuration Data:
(
        verbose => '2',
);

                      Is make installed...................................yes.
inter::lex -          Is lex installed...
Setting Configuration Data:
(
        flex_version => undef,
);

Setting Configuration Data:
(
        lex => 'echo',
);

Setting Configuration Data:
(
        verbose => '2',
);

                      Is lex installed................................skipped.
inter::yacc -         Is yacc installed...
Setting Configuration Data:
(
        bison_version => undef,
);

Setting Configuration Data:
(
        yacc => 'echo',
);

Setting Configuration Data:
(
        verbose => '2',
);

                      Is yacc installed...............................skipped.
auto::gcc -           Is your C compiler actually gcc...cl -nologo -GF -W4 -MD -Zi -DNDEBUG -Ox -GL -fp:precise -DWIN32 -D_CONSOLE -DNO_STRICT -DWIN64 -DCONSERVATIVE -DUS
E_SITECUSTOMIZE   -I./include -c test_3108.c
test_3108.c
test_3108.c(17) : warning C4100: 'argv' : unreferenced formal parameter
test_3108.c(17) : warning C4100: 'argc' : unreferenced formal parameter
cl -nologo -nodefaultlib -debug -opt:ref,icf -ltcg    -machine:AMD64 test_3108.obj  -out:test_3108.exe  kernel32.lib ws2_32.lib msvcrt.lib oldnames.lib advapi32.lib
cl : Command line warning D9035 : option 'o' has been deprecated and will be removed in a future release
cl : Command line warning D9035 : option 'o' has been deprecated and will be removed in a future release
cl : Command line warning D9002 : ignoring unknown option '-nodefaultlib'
cl : Command line warning D9002 : ignoring unknown option '-debug'
cl : Command line warning D9002 : ignoring unknown option '-ltcg'
cl : Command line warning D9002 : ignoring unknown option '-machine:AMD64'
test_3108.obj : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
Generating code
Finished generating code
.\test_3108.exe
Der Befehl ".\test_3108.exe" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.

step auto::gcc died during execution: Can't run the test program:  at config/auto/gcc.pm line 42.

 at Configure.pl line 96.
Command failed (status 256): C:\Perl64\bin\perl.exe Configure.pl --verbose=2 --fatal --icushared="C:\icu\lib64\icudt.lib C:\icu\lib64\icuuc.lib" --icuheaders=C:\icu\inclu
de --optimize --prefix=C:/nqp/install
```

**Comment by rurban** [comment]
Which Visual Studio version is this? I couldn't test all for the last win64 fixes.

**Comment by FROGGS** [comment]
```
C:\nqp\parrot>cl
Microsoft (R) C/C++ Optimizing Compiler Version 17.00.50727.1 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.
```

So it is Visual Studio Express 2012.

**Comment by rurban** [comment]
Ah, I tried only cl 16

**Comment by rurban** [comment]
Fixed in master (7.3.0)
